compiler_gym/spaces/reward.py

```diff
@@ -99,7 +99,7 @@ def reset(self, benchmark: str) -> None:
 
     def update(
         self,
-        action: int,
+        actions: List[int],
         observations: List[ObservationType],
         observation_view: "compiler_gym.views.ObservationView",  # noqa: F821
     ) -> RewardType:
```

Comment:
I'm curious, why doesn't it return a list of rewards, one for each observation passed in? What are the semantics of that supposed to be?

Reply:
What's the difference between a 'ObservationSpaceSpec' and a 'Reward'? 
Why does this take in 'observations', rather than 'rewards'?

Reply:
> I'm curious, why doesn't it return a list of rewards, one for each observation passed in? What are the semantics of that supposed to be?

`Reward.update()` is used to compute and return an incremental reward. It receives as inputs a list of observations that it can derive this reward from. E.g. for LLVM codesize, it will use the `IrInstructionCount` observation, but it could be any combination of observations. For example it could combine code size, instruction count, and runtime measurements. I will clarify the docs.